### PR TITLE
Refactors index.js

### PR DESCRIPTION
**Pull request by jeremy-then**
* Refactors index.js to include better checks for parameters
* General refactors
* Added extra tests

### CODE REVIEW

**Review by marcos-iov — APPROVED** [review]
Well done!

File: index.js [hunk 1]
```diff
@@ -21,18 +33,27 @@ const getRedeemScriptFromBtcPublicKeys = (btcPublicKeys) => {
...
         throw new Error(ERROR_MESSAGES.INVALID_P2SH_ERP_PUBLIC_KEYS);
     }
-    if (!csvValue || typeof csvValue !== 'number' || csvValue <= 0 || csvValue > MAX_CSV_VALUE) {
+
+    if (!Number.isInteger(csvValue) || csvValue < 1 || csvValue > MAX_CSV_VALUE) {
         throw new Error(ERROR_MESSAGES.INVALID_CSV_VALUE);
     }
 
-    csvValue = signedNumberToHexStringLE(csvValue);
+    const csvLeHexValue = signedNumberToHexStringLE(csvValue);
```
Comment (julia-zack) [inline index.js:56]: i'd put LE instead of Le, just to be consistent

File: index.js [hunk 2]
```diff
@@ -21,18 +33,27 @@ const getRedeemScriptFromBtcPublicKeys = (btcPublicKeys) => {
...
 
     const defaultRedeemScript = getPowpegRedeemScript(powpegBtcPublicKeys).toString('hex');
```
Comment (julia-zack) [inline index.js:58]: this is not necessary. you are doing the check inside getPowpegRedeemScript in line 45

Comment (julia-zack) [reply index.js:58]: I mean, this should be just 
const defaultRedeemScript = getRedeemScriptFromBtcPublicKeys(powpegBtcPublicKeys).toString('hex');

File: index.js [hunk 3]
```diff
@@ -21,18 +33,27 @@ const getRedeemScriptFromBtcPublicKeys = (btcPublicKeys) => {
     return bitcoin.payments.p2ms({ m: parseInt(defaultPubkeys.length / 2) + 1, pubkeys: defaultPubkeys }).output;
 };
 
+/**
+ * 
+ * @param {String[] | Buffer[]} powpegBtcPublicKeys 
+ * @param {String[] | Buffer[]} p2shErpBtcPublicKeys
```
Comment (julia-zack) [inline index.js:39]: I think these names are a bit confusing. I would rename _p2shErpBtcPublicKeys_ to _emergencyBtcPublicKeys_ or _erpBtcPublicKeys_.

**Review by julia-zack — APPROVED** [review]
Just one more comment but lgtm :)

File: utils.js [hunk 4]
```diff
@@ -38,38 +40,33 @@ const isValidNetwork = (network) => {
...
-        // copies it and adds it to `byteArray`, and keeps doing that until all bytes are copied.
-        // When `i` is 0, then `i * COUNT_OF_BITS_IN_BYTE` is 0 and hence `number` does not git shifted and we copy the first byte.
-        const nextByte = (number >> (i * COUNT_OF_BITS_IN_BYTE)) & oneByteMask;
-        byteArray.push(nextByte);
+    for(let i = 0; i < bytesCount; i++) {
+        const nextByteStartIndex = i * COUNT_OF_BITS_IN_BYTE;
+        const oneByteMaskMovedAtNextByteStartIndex = ONE_BYTE_MASK << nextByteStartIndex;
+        const nextByteCopyAtOriginalPosition = number & oneByteMaskMovedAtNextByteStartIndex;
+        const nextByteCopyMovedAtStart = nextByteCopyAtOriginalPosition >> nextByteStartIndex;
+        leBytesArray.push(nextByteCopyMovedAtStart);
```
Comment (julia-zack) [inline utils.js:57]: i'd put LEbytesArray instead of leBytesArray, to be consistent

File: utils.js [hunk 5]
```diff
@@ -38,38 +40,33 @@ const isValidNetwork = (network) => {
...
-    // then the bit is on.
-    const mostSignificantBitIsOn = (number & (1 << mostSignificantBitPosition)) > 0;
+    const mostSignificantBitPosition = bytesCount * COUNT_OF_BITS_IN_BYTE - 1;
+    const oneBitMaskMovedAtMostSignificantBitPosition = ONE_BIT_MASK << mostSignificantBitPosition;
+    const mostSignificantBitCopy = number & oneBitMaskMovedAtMostSignificantBitPosition;
+    const mostSignificantBitIsOn = mostSignificantBitCopy > 0;
 
     if (mostSignificantBitIsOn) {
-        byteArray.push(0);
+        leBytesArray.push(0);
```
Comment (julia-zack) [inline utils.js:66]: same here

File: utils.js [hunk 6]
```diff
@@ -38,38 +40,33 @@ const isValidNetwork = (network) => {
...
     }
 
-    return Buffer.from(byteArray).toString("hex");
+    return Buffer.from(leBytesArray).toString("hex");
```
Comment (julia-zack) [inline utils.js:69]: same here

**Review by julia-zack — APPROVED** [review]

File: utils.js (hunk 4 above)
Comment (nathanieliov) [reply utils.js]: It's weird to see variables starting with uppercase. I'd rather change the name of the method to use camel case. Or maybe just use the words LE stands for.

**Review by nathanieliov — APPROVED** [review]
LGTM

**Pull request merged**